Source/Core/DolphinQt/Debugger/TraceWidget.cpp

```diff
@@ -0,0 +1,952 @@
+}
+
+void TraceWidget::UpdateBreakpoints()
+{
+  // Leave the recorded end intact.
+  if (m_record_trace->isChecked())
+  {
+    for (int i = m_record_stop_addr->count(); i > 1; i--)
+      m_record_stop_addr->removeItem(1);
+  }
+  else
+  {
+    m_record_stop_addr->clear();
+  }
+  auto& power_pc = m_system.GetPowerPC();
+  auto& breakpoints = power_pc.GetBreakPoints();
+  Core::CPUThreadGuard guard(m_system);
+
+  int index = -1;
+
+  for (const auto& bp : breakpoints.GetBreakPoints())
+  {
+    QString instr =
+        QString::fromStdString(power_pc.GetDebugInterface().Disassemble(&guard, bp.address));
+    instr.replace(QStringLiteral("\t"), QStringLiteral(" "));
+    m_record_stop_addr->addItem(
+        ElidedText(QStringLiteral("%1 : %2").arg(bp.address, 8, 16, QLatin1Char('0')).arg(instr)),
+        bp.address);
+    index++;
+  }
+
+  // User typically wants the most recently placed breakpoint.
+  if (!m_record_trace->isChecked())
+    m_record_stop_addr->setCurrentIndex(index);
+}
+
+void TraceWidget::InfoDisp()
+{
+  // i18n: Here, PC is an acronym for program counter, not personal computer.
+  auto* info = new QTableWidgetItem(QStringLiteral(
+      "Used to \"track the value in a target register or memory address and its uses.\n\n"
+      "Record Trace: Records each executed instruction while stepping from PC to selected "
+      "Breakpoint. Required before tracking a target. If backtracing, set PC to how far back you "
+      "want to trace to and breakpoint the instruction you want to trace backwards.\n\n"
+      "Register: Input examples: r5, f31, use f for ps registers or 80000000 for memory. Only "
+      "takes one value at a time. Leave blank to view complete code path.\n\n"
+      "Starting Address: Used to change range before tracking a value. Record Trace's starting "
+      "address is always the PC. Can change freely after recording trace.\n\n"
+      "Ending breakpoint: Where the trace will stop. If backtracing, should be the line you want "
+      "to backtrace from.\n\n"
+      "Backtrace: A reverse trace that shows where a value came from, the first output line is "
+      "the "
+      "most recent executed.\n\n"
+      "Verbose: Will record all references to what is being tracked, rather than just where it "
+      "is "
+      "moving to or from.\n\n"
+      "Reset on loopback: Will clear the trace if starting address is looped through, ensuring "
+      "only the final loop to the end breakpoint is recorded.\n\n"
+      "Change Range: Change the start and end points of the trace for tracking. Loops may make "
+      "certain ranges buggy.\n\n"
+      "Track target: Follows the register or memory value through the recorded trace. You don't "
+      "have to record a trace multiple times if the first trace recorded the area of code you "
+      "need. You can change any value or option and press track target again. Changing the "
+      "second "
+      "breakpoint will let you backtrace from a new location."));
```

Comment:
Some names mentioned here don't seem to match the UI interface, I don't see a `verbose` option for instance.